### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by TOTHE34** [Jean-Romain/tabliato#6] [Q&A]
Bonjour, 

Je n'arrive pas à installer tabliato, j'ai bien installé lilypond mais je ne trouve pas de fichier exécutables dans l'archive. Je ne suis pas très doué en code mais je suis très intéressé par le logiciel. Merci beaucoup !

**Comment by Jean-Romain** [comment]
Il n'y a présentement aucun executable. Il faut compiler le code soit même. Normalement ce n'est pas très compliqué. Vous allez dans le dossier `tabliato` vous ouvrez un terminal et

```
sudo apt-get install qtbase5-dev lilypond timidity timidity-interfaces-extra freepats libpoppler-qt5-dev qtmultimedia5-dev
qmake
make
```

Si vous avez besoin de plus d'aide dites moi. Je peux vous founir un `.deb` aussi. Il y a aussi un `makefile` pour faire un `.deb` dans le dossier `debian`. A une epoque il y avait même une version windows. Mais étant le seul utilisateur j'ai abandonné le fastidieux travail de packaging

**Reply by TOTHE34** [reply]
Merci pour votre réponse, c'est justement une version windows qu'il me faudrait, ce ne sera donc pas possible ?

**Reply by Jean-Romain** [reply]
Techniquement c'est possible en compilant soit même. Mais sous windows bon courage... Après Windows peut faire tourner des programmes linux. Il y a un utilisateur qui l'a fait ici https://github.com/Jean-Romain/tabliato/issues/4 mais je n'en sais pas plus. Si windows est capable d'installer et de rouler des paquets debian c'est surement le plus simple. Je peux essayer de produire un binaire Windows mais ca va me prendre quelques semaines. Je tag @agentils peut être qu'il pourra nous expliquer s'il voit ce message.

**Reply by TOTHE34** [reply]
Ça marche, merci pour ces informations. Je vais essayer de me débrouiller, ne vous lancez pas dans plus de travail juste pour moi.

**Reply by Jean-Romain** [reply]
En fait ca me gène pas du tout. Si il n'y a pas de version Windows c'est parce que j'ai aucune raison d'y passer du temps et personne pour tester. Mais si il y a une demande et la possibilité de faire tester alors c'est cool je veux bien le faire. Ca servira aux prochains utilisateurs. Par contre je ne peux pas le faire vite.

**Reply by Jean-Romain** [reply]
Cherchez ce que c'est que c'est que WSL et si on peut installer des paquet debian .deb. Si oui alors je vous donnerai un .deb a installer.

**Comment by agentils** [comment]
Hello,
De mon côté j'ai compilé les sources dans WSL (de mémoire il faut WSL2 pour
faire fonctionner l'interface graphique). Il manquait quelques paquets mais
j'ai réussi à trouver quoi ajouter avec une recherche Google rapide.
Je n'ai pas noté les étapes à suivre mais je peux chercher dans
l'historique si besoin.
Du coup un fichier deb aurait légèrement accéléré le process en effet, mais
ce n'est pas insurmontable.
Bon courage
Aurélien

**Comment by Jean-Romain** [comment]
J'ai uploadé un `.deb` [ici](https://drive.google.com/file/d/1WPF57FfnYAZYOfnXmMKuDsrTW78YI6tk/view?usp=sharing). Je vais essayer de produire un `.exe` pour windows aussi.

**Comment by Jean-Romain** [comment]
Bon je pense que c'est mort pour Windows. Le framework Qt (pour faire des fenêtre) a évolué depuis les années et semble ne plus être open source. Il faut se créer un compte pour obtenir Qt sous Windows... Impossible de trouver une ancienne version libre de `Qtcreator`.

**Comment by TOTHE34** [comment]
Merci pour tout ça ! J'ai bien installé WSL2 et j'ai ma console Debian opérationnelle mais je ne parviens pas encore à installer le programme à partir du .deb,

**Reply by Jean-Romain** [reply]
Ben alors là, moi je peux pas aider désolé. Je ne sais pas même pas si c'est possible et je ne connaissant pas WSL avant le message de @agentils . On va demander a @agentils ou sinon a google. J'ai essayé de faire une version windows hier mais impossible d'installer quoique soit qui fonctionne sous Windows. Il y a 8 ans c'était pas si compliqué et j'avais fait une version windows. Aujourd'hui c'est rendu infernal et je n'arrive a rien avec ma machine virtuelle sous windows 7.

**Comment by TOTHE34** [comment]
Ne vous en faites pas je vais finir par m'en sortir ! J'ai réussi à l'installer, mais je n'arrive pas encore à la faire tourner.
tote@LAPTOP-BHBV3483:~$ tabliato
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
Aborted 

Apparemment il faut que j'installe un plugin.

**Reply by Jean-Romain** [reply]
Bon courage. Si vous arrivez au bout du processus merci de poster la solution ça pourrait servir. En attendant moi je vais essayer de me trouver un ordi avec windows 10 pour voir si il y a moyen de faire qqch a moyen terme.

**Comment by Jean-Romain** [comment]
Excellente nouvelle @TOTHE34 @agentils !! Après de nombreuses tentatives et essais infructueux et bugs invraisemblables j'ai réussi a produire une version Windows. Pour l'instant ce n'est pas fonctionnel pour vous mais je pense avoir fait 80% du chemin. Le 20% restant ne devrait pas poser trop de problèmes j'espère.

![tabliato-win](https://user-images.githubusercontent.com/3872279/159194751-4cd39926-5478-4949-8cc8-a338f6ff4917.PNG)

**Reply by Jean-Romain** [reply]
J'ai fait 95% du chemin. J'ai encore:

1. un petit problème d’icônes qui ne s'affichent pas
2. un problème de lecture de la musique générée qui ne s'update pas
3. un problème de soundfound c'est a dire que la musique générée est générée avec du piano et pas de l'accordéon...

Pour 2 et 3 à la limite ça peut marcher sans cette fonctionnalité mais moi je m'en sers beaucoup pour écouter mes erreurs.

**Comment by Chris-afk71** [comment]
Bonjour,
Je suis également intéressé par une installation sous Windows. 
Si vous avez une procédure d'installation, pourrais je en profiter ? 
Ca fait un moment que j'essaie de l'installer sans succès...
Merci d'avance.
Chris

**Reply by Jean-Romain** [reply]
Ca va venir j'y suis presque.

**Reply by Jean-Romain** [reply]
Si ca se passe bien je devrais pouvoir proposer une version windows en fin de semaine

**Comment by Jean-Romain** [comment]
Ca avance. J'estime avoir fait 98% du chemin

- [x] Compilation du logicel sous windows
- [x] Production d'une tablature
- [x] Génération de la musique
- [ ] Génération de la musique avec de l'accordéon pas du piano
- [x] Affichage des icones
- [x] Lecture de la musique
- [ ] Sauvegarde et lecture des settings

**Comment by Jean-Romain** [comment]
Voici une version windows. Tout ne marche pas encore mais ca doit générer au moins des partitions en pdf. Je vais continuer de travailler a faire fonctionner ce qui ne marche pas.

1. Installer [lilypond](https://lilypond.org/windows.html)
2. Télécharger [l'archive de tabliato](https://drive.google.com/file/d/1Yjiu6WzILq4IQSxpQvfHu7w9jcXTfAKn/view?usp=sharing)
3. Dezipper l'archive
4. Lancer `tabliato.exe`

Merci pour vos retour.

**Reply by TOTHE34** [reply]
Bonjour, merci pour tous vos efforts ! C'est génial, j'essaye ça dans l'après midi !!

**Comment by agentils** [comment]
Bonjour,
Bravo à toi Jean-Romain, je vais tester cela dès que possible et je te
ferai un retour.
Aurélien

**Comment by Chris-afk71** [comment]
Bonjour,
J'ai un message d'erreur au lancement "This application failes to start because not QT platform plugin could be initialized."
J'ai installé Python 3.10 mais la commande "pip install pyqt5-tools" n'arrive pas au bout...
Ca n'est peut être la bonne solution que j'utilise.
Chris

**Reply by Jean-Romain** [reply]
Il n'y a pas de python dans tabliato ce n'est que du C++. Dites moi exactement ce que vous avez fait, dans quel dossier vous avez extrait l'archive, comment vous avez lancé le logiciel, quelle version de windows vous utilisez. Bref tout ce qui peut m'aider. J’essaierai de reproduire ce soir dans une machine virtuelle indépendante.

Pour info je n'utilise plus windows depuis plus de 10 ans et je connais très mal le système donc j'ai un peu de difficulté a savoir a l'avance quels problèmes on va rencontrer. On va faire marcher le logiciel ensemble avec vos retours si vous voulez bien. Moi je suis  motivé ça faisait des années que voulais produire une version windows mais je n'avais aucune motivation a me lancer la dedans sans raison.

**Comment by Chris-afk71** [comment]
Merci beaucoup, ci dessous les principales info je pense : 

Windows 10 Entreprise Version 20H2 Build 19042.1586. 
Appli installés : 
Office Pro 2021
Microsoft Visual C++ x64 Redistributable 2008, 2010, 2012, 2013, 2015-2019
Microsoft Visual C++ x86 Redistributable 2008, 2010, 2012, 2013, 2015-2019

Les fichiers sont extrait dans un répertoire C:\tabliato\
Lancement par double clic

**Comment by TOTHE34** [comment]
J'ai exactement le même problème que @Chris-afk71 et j'avais le même message lorsque j'essayais de le lancer avec wsl. 
Windows 10 famille 
Version 1903 - 18362.1139
C:\tabliato

J'essaie de trouver des solutions mais pour l'instant rien ne fonctionne

**Comment by Jean-Romain** [comment]
Bon j'ai essayé dans une machine virtuelle indépendante de celle avec laquelle j'ai développé et j'ai le même problème.

**Comment by Jean-Romain** [comment]
Voici une autre tentative de version pour windows [ici](https://drive.google.com/file/d/1IWceGv7TvYe-6T5QOdz-9iG2frRwOjLg/view?usp=sharing). Cette version semble ne pas fonctionner sous windows 7 mais semble fonctionner sous windows 10. Je dois utiliser 3 versions de windows différentes pour faire mes tests:

1. Windows 10 pour le développement et la compilation. Comme tout les outils de développement sont installé tabliato fonctionne toujours bien
2. Windows 7 pour les tests sans outils de développement pour reproduire le cas d'un utilisateur normal. Le problème c'est que ca ne fonctionne pas sous W7 et que je n'ai pas encore trouvé pourquoi
3. Une autre machine sous windows 10 mais sur laquelle je ne peux pas installer `lilypond` donc je n'ai pas pu faire un test complet. Mais au moins `tabliato` ne crash pas comme sous W10.

Voilà voilà, c'est pour ca que je developpe sous linux :wink:  Allez je croise les doigts :crossed_fingers:

**Comment by Jean-Romain** [comment]
Bon, j'ai trouvé pourquoi ça ne marchait pas. Laissez tomber le lien précédent voici une version testée dans un environnent vierge sous W7. 

[tabliato-win32-alpha](https://drive.google.com/file/d/1xbtpUfHCfa1WRxt4edFGTICxy7FCGMnW/view?usp=sharing)

Je suis très confiant qu'on a une bonne base qui va marcher là. Par contre le rendu audio et la lecture audio ne fonctionnent toujours pas mais c'est un détail.

**Comment by Chris-afk71** [comment]
Génial ça fonctionne 👍 
Beau boulot, on peut commencer à jouer avec.

**Comment by Chris-afk71** [comment]
J'adore son utilisation simple et logique.
La prise en main est super facile.
J'ai noté que les accents que l'on met dans information à gauche (Titre) ne passe pas dans la prévisualisation (suppression de la lettre).
je n'arrive pas à faire passer sur une page 2 (avec des \break)
quand je fait une \alternative de deux mesures il me met une répétition entre la mesure 1 et 2 de la répétition... ou je l'ai mal écrit...
J'attends la partie audio avec impatience :)
Merci encore.

**Reply by Chris-afk71** [reply]
![image](https://user-images.githubusercontent.com/102148143/159903749-262a5335-c720-40e1-ba05-3bfcfa670ca2.png)
![image](https://user-images.githubusercontent.com/102148143/159903859-d3ccddbf-ed53-4fa9-9260-52f77cbd8810.png)

**Reply by Jean-Romain** [reply]
Là comme ca je dirai qu'il y a deux G de trop dans la partie principale.

**Comment by Jean-Romain** [comment]
> J'adore son utilisation simple et logique. La prise en main est super facile.

Merci j'apprecie

> J'ai noté que les accents que l'on met dans information à gauche (Titre) ne passe pas dans la prévisualisation (suppression de la lettre).

Pas une grosse surprise pour moi ca. Mais s'il vous plait ouvrez un rapport de bug (onglet [issues](https://github.com/Jean-Romain/tabliato)) par problème/question. Sinon je n'arriverai pas à gérer. Pas de screenshot de code seulement des copier coller (je dois pouvoir copier coller moi aussi pour tester). Et un screenshoot du problème si nécessaire. Merci.

> je n'arrive pas à faire passer sur une page 2 (avec des \break) quand je fait une \alternative de deux mesures il me met une répétition entre la mesure 1 et 2 de la répétition.

Pareil ouvrez une nouvelle *issue* indépendante. Merci.

> J'attends la partie audio avec impatience :)

Oui c'est assez satisfaisant d'écouter ce qu'on a écrit :smile:. Par contre je vais avoir un peu de mal à résoudre le problème car pour le moment je ne comprend pas pourquoi ça ne marche pas... mais bon j'y arriverai.

Aussi sentez vous libre de suggérer des améliorations. J'ai créé tabliato pour moi et tabliato marche bien pour moi mais je suis content d'avoir enfin extérieurs pour continuer le dev

**Reply by Chris-afk71** [reply]
ok pb remonté dans la bonne partie pour les accents. 
Le reste est réglé c'est moi qui l'utilisait mal.

**Comment by agentils** [comment]
Bonjour,
La commande \break permet de passer à la ligne suivante. Il y a
probablement une commande pour changer de page mais je ne la connais pas,
il faudrait regarder dans la doc de Lilypond.
Pour \alternative, je pense que c'est un problème d'accords (il faut
indiquer les accords pour chaque mesure comme ce que tu fais pour le reste).
Concernant les accents ce n'est probablement pas normal.
Bon courage, la syntaxe n'est pas toujours évidente mais c'est très
pratique une fois qu'on s'est familiarisé avec tout ça.
Aurélien

Le jeu. 24 mars 2022 à 12:09, Chris-afk71 ***@***.***> a
écrit :

**Reply by Jean-Romain** [reply]
> Bon courage, la syntaxe n'est pas toujours évidente mais c'est très pratique une fois qu'on s'est familiarisé avec tout ça.

Oui tout à fait. C'est lourdement basé sur LaTeX et complètement sur lilypond. La syntaxe de lilypond est imbuvable mais une fois qu'on s'y ai habitué c'est incroyablement efficace et versatile.

> Concernant les accents ce n'est probablement pas normal.

En effet. Problème d'encodage sous windows je vais regarder ca.

> probablement une commande pour changer de page mais je ne la connais pas,

Moi non plus. Mais il change de page tout seul quand la page est pleine. Après si on cherche dans la doc de lilypond on trouve `\pageBreak`. A tester

**Comment by TOTHE34** [comment]
Merci beaucoup ça fonctionne très bien pour moi, en revanche je n'arrive pas à exporter en PDF. Est-ce qu'il y a une manip particulière ? Lorsque je clique sur "outils" et  "exporter en pdf " l'option semble se cocher et se décocher mais il ne se passe rien.

**Reply by Jean-Romain** [reply]
Ca ressemble a un bug. Normalement l'option est cochée et le `.pdf` va avec le fichier `.dtb` quand on clique sur <kbd>compiler</kbd>.

**Reply by Jean-Romain** [reply]
Comme je l'ai dit avant s'il vous plaît ouvrez une [issue](https://github.com/Jean-Romain/tabliato/issues) différente pour chaque question ou  problème sinon c'est ingérable pour moi merci. Et pensez a décrire clairement ce que vous avez fait/essayé/vu que je puisse refaire la même chose et trouver le bug ou simplement vous dire comment faire autrement dans le meilleur des cas

**Comment by Jean-Romain** [comment]
On va fermer cette discussion. Le sujet était: *installation sous windows*. Le sujet est clos, j'ai produit un binaire windows qui marche. Je vous invite a suivre ce qui se passe dans les autres fils de discussion pour voir la suite des résolutions de bug et pour poser vos questions auxquels il me fera plaisir de répondre.

Vous pouvez cliquer sur <kbd>Watch</kbd> en haut a droite pour être notifié des nouveaux messages.
